### DISCUSSION TITLE
We need elaborate profiling for ABACUS version 2.2.0, please collect as many case as you can!

### DISCUSSION THREAD

**Discussion by dyzheng** [deepmodeling/abacus-develop#716] [Show and tell]
## Describe Current Status and Possible Solution

We can seperate the time consumed by the program into several parts collected in the end of screem or OUT.suffix/running_scf.txt outputs.

For example:

|CLASS_NAME---------|NAME---------------|TIME(Sec)-----|CALLS----|AVG------|PER%-------
Gint_k                          vlocal                        46.585             10             4.7            33        %

We can focus on PER% > 1% and mark NAME, CALLS, AVG;
another significant feature is cores in this calculation.
Better to note what type of machine and note which compiler.

**Comment by dyzheng** [comment]
units: second
<html xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40">
<head>

<meta name=Generator content="Microsoft Excel">

<style>

</style>
</head>
<body>



oneapi+8269CY | scalapack对角化 | vlocal格点积分矩阵元 | nonlocal矩阵元 | S矩阵元 | sum_bands | cal_dk_k | cal_rho_k | mix_rho
-- | -- | -- | -- | -- | -- | -- | -- | --
128Si | 2.4 | 4.7 | 7.6 | 0.67 | 2.9 | 0.36 | 2.3 | 1.1
256Si | 22 | 9.5 | 17 | 1.5 | 7.9 | 1.9 | 4.9 | 1.2
512Si | 2.00E+02 | 20 | 39 | 3.2 | 28 | 11 | 10 | 1.2
GNU+5600x | elpa对角化 | vlocal格点积分矩阵元 | nonlocal矩阵元 | S矩阵元 | sum_bands | cal_dk_k | cal_rho_k | mix_rho
128Si | 1.6 | 22 | 3.4 | 0.37 | 11 | 0.25 | 10 | 0.79
256Si | 14 | 46 | 7.7 | 23 | 23 | 1.3 | 21 | 0.7
512Si | 1.10E+02 | 1.00E+02 | 17 | 1.9 | 59 | 6.9 | 47 | 0.74



</body>

</html>

**Comment by caic99** [comment]
To optimize bottleneck, I suggest we don't need so much samples. The bottlenecks should be consistent across samples, while the algorithm is static. A few profiling cases are sufficient to guide our optimize direction.

**Comment by pxlxingliang** [comment]
The format of time in running_scf.log is not scientific and only 2 significant digits are reserved, it is better to make it same with that in OUTPUT.

**Comment by dyzheng** [comment]
Some timer of function has error, we should keep repairing them.

**Comment by dyzheng** [comment]
> To optimize bottleneck, I suggest we don't need so much samples. The bottlenecks should be consistent across samples, while the algorithm is static. A few profiling cases are sufficient to guide our optimize direction.

The new target is to elaborate profiling for several cases.

**Comment by caic99** [comment]
> > To optimize bottleneck, I suggest we don't need so much samples. The bottlenecks should be consistent across samples, while the algorithm is static. A few profiling cases are sufficient to guide our optimize direction.
> 
> The new target is to elaborate profiling for several cases.

That's good. Will you provide such cases? @dyzheng

**Comment by mohanchen** [comment]
> The bottlenecks should be consistent across samples

Of course not, please learn some DFT knowledge when you have time.

**Comment by mohanchen** [comment]
> Some timer of function has error, we should keep repairing them.

This should be another issue.

**Comment by mohanchen** [comment]
> The format of time in running_scf.log is not scientific and only 2 significant digits are reserved, it is better to make it same with that in OUTPUT.

This is another issue.
